scripts/eslint/rules/no-array-mutation-method-returns.cjs

```diff
@@ -0,0 +1,146 @@
+const { ESLintUtils } = require("@typescript-eslint/utils");
+const { createRule } = require("./utils.cjs");
+const { getConstrainedTypeAtLocation, isTypeArrayTypeOrUnionOfArrayTypes } = require("@typescript-eslint/type-utils");
+
+/**
+ * @import { TSESTree } from "@typescript-eslint/utils"
+ */
+void 0;
+
+module.exports = createRule({
+    name: "no-array-mutation-method-returns",
+    meta: {
+        docs: {
+            description: ``,
+        },
+        messages: {
+            noSideEffectUse: `This call to {{method}} appears to be unintentional as it appears in an expression position. Sort the array in a separate statement, or use slice/{{toMethod}} to ensure a copy is made.`,

```

Comment:
This message with `toMethod` is slightly untrue because we only have `toSorted`, not `toReversed` and `toSpliced` (without defining polyfills for ourselves).